modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -1564,21 +1564,21 @@ public CommerceOrder updatePaymentStatusAndTransactionId(
 			String transactionId)
 		throws PortalException {
 
-		CommerceOrder commerceOrder = commerceOrderPersistence.findByPrimaryKey(
-			commerceOrderId);
+		CommerceOrder oldCommerceOrder =
+			commerceOrderPersistence.findByPrimaryKey(commerceOrderId);
 
-		int previousPaymentStatus = commerceOrder.getPaymentStatus();
+		CommerceOrder newCommerceOrder = oldCommerceOrder;
```

Comment:
@pedro-oliveira446 

use `originalCommerceOrder.clone();` to avoid reference issues.